# OrthoFinder Orthogroup Summary Tool

[![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.18289369.svg)](https://doi.org/10.5281/zenodo.18289369)

A lightweight R script to summarize OrthoFinder orthogroup results
into a concise TSV file for downstream comparative genomics analyses.


---

## Background

OrthoFinder provides rich orthogroup outputs, but many downstream analyses
require only numerical summaries rather than full gene lists.

This tool focuses on:
- Clean numerical aggregation
- Minimal dependencies (base R only)
- Easy integration into analysis pipelines

---

## Example_dataset

The following dataset was used for testing:

- Species: Xenopus laevis,Xenopus tropicalis
- Source: Ensembl / NCBI RefSeq
- Data type: fasta,gff

---
## Testdata
This repository does not include test data due to file size and license considerations.

Publicly available FASTA and gff files can be downloaded from Ensembl or NCBI.
Example commands are provided below.

# Xenopus laevis
```
wget https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/017/654/675/GCF_017654675.1_Xenopus_laevis_v10.1/GCF_017654675.1_Xenopus_laevis_v10.1_genomic.fna.gz
wget https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/017/654/675/GCF_017654675.1_Xenopus_laevis_v10.1/GCF_017654675.1_Xenopus_laevis_v10.1_genomic.gff.gz
``` 
# Xenopus tropicalis
```
wget https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/004/195/GCF_000004195.4_UCB_Xtro_10.0/GCF_000004195.4_UCB_Xtro_10.0_genomic.fna.gz
wget https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/004/195/GCF_000004195.4_UCB_Xtro_10.0/GCF_000004195.4_UCB_Xtro_10.0_genomic.gff.gz
```

---

## Input

This script expects the following files generated by OrthoFinder:

- `Orthogroups.GeneCount.tsv`
- `Orthogroups.tsv`

---

## Usage

Run the script using `Rscript`:

```bash
Rscript scripts/summarize_orthogroups.R \
  --gene_count Orthogroups.GeneCount.tsv \
  --orthogroups Orthogroups.tsv \
  --species XL,XT \
  --out_prefix XL_XT

To see all available options:

```bash
Rscript scripts/summarize_orthogroups.R --help

Output
* <out_prefix>.summary.tsv
The output file contains:
* Orthogroup ID
* Total gene counts
* Per-species gene counts
* Per-species gene member lists
This TSV file can be directly imported into R, Python, or spreadsheet software for downstream analyses.
An example output file is available in the `examples/` directory.

Design Philosophy
This tool intentionally does NOT:
* Perform statistical testing
* Generate plots
* Modify OrthoFinder outputs
The goal is to provide a clean numerical summary that users can flexibly process in downstream workflows.

Intended Users
* Comparative genomics researchers
* Evolutionary biologists
* Bioinformatics pipeline developers

Requirements
* R (>= 3.6)
* No additional packages required

Citation
If you use this script in your research, please cite:
* Emms, D. M. & Kelly, S. (2019) OrthoFinder

License
MIT License Copyright (c) 2026 Ryota Ichikawa
